lib/ui/painting/image_descriptor.cc

```diff
@@ -128,4 +163,36 @@ void ImageDescriptor::instantiateCodec(Dart_Handle codec_handle,
   }
   ui_codec->AssociateWithDartWrapper(codec_handle);
 }
+
+sk_sp<SkImage> ImageDescriptor::image() const {
+  if (platform_image_generator_) {
+    SkBitmap bitmap;
+    if (!bitmap.tryAllocPixels(image_info_)) {
+      FML_LOG(ERROR) << "Failed to allocate memory for bitmap of size "
+                     << image_info_.computeMinByteSize() << "B";
+      return nullptr;
+    }
+
+    const auto& pixmap = bitmap.pixmap();
+    if (!platform_image_generator_->getPixels(
```

Comment:
Btw, we recently added SkImageGenerator::getPixels(const SkPixmap&pm), so you can simplify this line (and others) to just
```suggestion
    if (!platform_image_generator_->getPixels(pixmap)) {
```

Reply:
Done here and in another spot.